.github/workflows/auto-approve.yml

```
@@ -1,13 +1,72 @@
-name: Auto approve
-
+name: auto approve and merge and tidy
 on:
-  pull_request
+  pull_request:
+    paths:
+      - ".github/workflows/gosum.yml"
+      - "go.mod"
+      - "go.sum"
 
+# job level conditions don't seem to work at the moment
+# https://github.community/t5/GitHub-Actions/Status-of-workflows-with-no-running-jobs/td-p/37160
+# which is why github.actor == 'dependabot[bot]' || github.actor == 'dependabot-preview[bot]' is
+# repeated over and over
 jobs:
-  auto-approve:
+  tidy:
+    name: run go mod tidy and updated
+    runs-on: ubuntu-latest
+    if: github.actor == 'dependabot[bot]' || github.actor == 'dependabot-preview[bot]'
+    steps:
+      - name: checkout
+        uses: actions/checkout@v1
+      - name: reattach HEAD to Head Ref
+        # b/c checkout action leaves in detached head state https://github.com/actions/checkout/issues/6
+        run: git checkout "$(echo ${{ github.head_ref }})"
+        if: github.head_ref != '' && (github.actor == 'dependabot[bot]' || github.actor == 'dependabot-preview[bot]')
```

```suggestion
        if: github.head_ref != ''
```

I think you only need the one piece. I am hoping there's a follow on effect from the above `if` statement that you moved on line 17.

---

let's find out!